lib/core/utils/preload-cssom.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Returns a then(able) queue of CSSStyleSheet(s)
+ * @param {Object} ownerDocument document object to be inspected for stylesheets
+ * @param {number} timeout on network request for stylesheet that need to be externally fetched
+ * @param {Function} getSheetFromTextFn a utility function to generate a style sheet from text
+ * @return {Object} queue
+ * @private
+ */
+function loadCssom(ownerDocument, timeout, getSheetFromTextFn) {
+	const q = axe.utils.queue();
+
+	Array.from(ownerDocument.styleSheets).forEach(sheet => {
+		if (sheet.disabled) {
+			return;
+		}
+
+		try {
+			if (sheet.cssRules) {
+				// accessing .cssRules throws for external (cross-domain) sheets, which is handled in the catch
+				sheet.isExternal = false; // isExternal is a flag to ascertain if a network request is necessary for cssom building
+				q.defer(resolve => resolve(sheet));
+			}
+		} catch (e) {
+			// external sheet -> make an xhr and q the response
+			q.defer((resolve, reject) => {
+				axe.imports
+					.axios({
+						method: 'get',
+						url: sheet.href,
+						timeout
+					})
+					.then(({ data }) => {
+						const sheet = getSheetFromTextFn(data, true); //second argument acts as > isExternal - true
```

Comment:
That second argument is useful. It's always external. More importantly, `isExternal` shouldn't be set in `getSheetFromTextFn` at all. It belongs in the same function that you set isExternal = false (this one).

Reply:
Solved!